clang/test/CodeGen/cheri/overloaded-builtins.c

```diff
@@ -436,71 +436,67 @@ void test_uintcap_t(__uintcap_t arg, __uintcap_t arg2) {
 }
 
 // PURECAP-LABEL: define {{[^@]+}}@test_null_constant
-// PURECAP-SAME: (i8 addrspace(200)* [[ARG:%.*]]) addrspace(200) #0
+// PURECAP-SAME: (i8 addrspace(200)* [[ARG:%.*]]) addrspace(200) #[[ATTR0]] {
 // PURECAP-NEXT:  entry:
 // PURECAP-NEXT:    [[TMP0:%.*]] = call i64 @llvm.cheri.cap.address.get.i64(i8 addrspace(200)* null)
-// PURECAP-NEXT:    [[TMP1:%.*]] = call i8 addrspace(200)* @llvm.cheri.cap.address.set.i64(i8 addrspace(200)* null, i64 1)
```

Comment:
We've lost the address.set here and offset.set below, I guess that's because of the early bail out in the address/offset set hooks for null pointers?

Reply:
yeah, `getTargetHooks().setPointerAddress()` handles null specially.